src/pkg/clients/core/image.go

```diff
@@ -29,12 +32,17 @@ func (c *client) ListAllArtifacts(project, repository string) ([]*modelsv2.Artif
 }
 
 func (c *client) DeleteArtifact(project, repository, digest string) error {
-	// /projects/{project_name}/repositories/{repository_name}/artifacts/{reference}
+	repository = encodeRepository(repository)
 	url := c.buildURL(fmt.Sprintf("/api/v2.0/projects/%s/repositories/%s/artifacts/%s", project, repository, digest))
 	return c.httpclient.Delete(url)
 }
 
 func (c *client) DeleteArtifactRepository(project, repository string) error {
+	repository = encodeRepository(repository)
 	url := c.buildURL(fmt.Sprintf("/api/repositories/%s/%s", project, repository))
 	return c.httpclient.Delete(url)
 }
+
+func encodeRepository(repository string) string {
```

Comment:
I remembered we had other places with the similar cases, did we define a utility encoding method before?

Is this repo encoding method align with the repo encoding way we discussed before?